### ISSUE TITLE
indexing optional propertys

### ISSUE THREAD

**Issue by creisle** [orientechnologies/orientdb#6828]
## OrientDB Version, operating system, or hardware.
- v2.2.11

## Operating System
- Linux

## Expected behavior and actual behavior

I have a class with multiple properties. One of the properties is optional. The others are not. I want to create a composite index on both of them

```
create class example extends v
create property example.name string
create property example.version integer
create property example.date date
alter property example.name  mandatory True
create index partial_unique on example (name, version, date) unique
```

however, now if I try to insert a record where version is null I get a null pointer exception. I would have expected it to allow me to insert things with nulls on optional, nullable properties still

the desired behaviour I am looking for is

```
create class example extends v
create property example.name string
create property example.version integer
create property example.date date
alter property example.name  mandatory True
create index partial_unique on example (name, version, date) unique
create vertex example content {'name': 'ex1'}
create vertex example content {'name': 'ex1', 'version': 1}
```

would be ok and then I would get a unique constraint error on the following

```
create vertex example content {'name': 'ex1'}
```

## Steps to reproduce the problem

```
create class example extends v
create property example.name string
create property example.version integer
create property example.date date
alter property example.name  mandatory True
create index partial_unique on example (name, version, date) unique
create vertex example content {'name': 'ex1'}
```

**Comment by luigidellaquila** [comment]
Hi @creisle 

I tried your steps on a fresh download of 2.2.11 but I was not able to reproduce it.
Am I missing something? Eg. was the DB created with a previous version?

Thanks

Luigi

**Comment by creisle** [comment]
Hi Luigi,

It wasn't created on an earlier version. I downloaded a fresh install this week. I did use the gui for the commands, perhaps that is the issue? also there is a typo in my comment above should be

```
create index partial_unique on example (name, version, date) unique
```

i've screenshotted the error I am seeing below

![image](https://cloud.githubusercontent.com/assets/4342641/19528059/695bdd0a-95df-11e6-8569-57a6c718c706.png)

it's worth noting that I think this only happens with the date property being null
